server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -195,6 +211,60 @@ public Status getStatus() {
         return null;
     }
 
+    /**
+     * Returns resource consumption of the task
+     */
+    public Map<String, List<TaskCompleteResourceInfo>> getResourceStats() {
+        return resourceStats;
+    }
+
+    public Map<String, Map<String, Long>> getTotalResourceStats() {
+        return new HashMap<String, Map<String, Long>>() {
+            {
+                put(TOTAL_RESOURCE_CONSUMPTION, new HashMap<String, Long>() {
+                    {
+                        Arrays.stream(TaskStats.values()).forEach(v -> put(v.toString(), getTotalResourceUtilization(v)));
+                    }
+                });
+            }
+        };
+    }
+
+    public long getTotalResourceUtilization(TaskStats taskStats) {
+        AtomicLong totalResourceConsumption = new AtomicLong();
```

Comment:
Using an AtomicLong just to be able to capture side effects from inside lambda functions seems like an anti-pattern to me. It implies there's some sort of concurrency here and also seems a bit like a hack to get around the compiler. I think you can rewrite this pretty simply to use regular for loops and a primitive long, like:

```
long totalResourceConsumption = 0L;
for (List<TaskCompleteResourceInfo> statsUtilList : resourceStats.values()) {
    for (TaskCompleteResourceInfo statsUtil : statsUtilList) {
        for (Map.Entry<TaskStatsType, TaskResourceInfo> entry : statsUtil.getResourceInfo().entrySet()) {
            if (!entry.getKey().isOnlyForAnalysis()) {
                totalResourceConsumption += entry.getValue().getStatsInfo().get(taskStats).getTotalValue();
            }
        }
    }
}
return totalResourceConsumption;
```

I think you can also write this using the Java stream API, but I find it a bit hard to read and might prefer the more verbose style above:

```
return resourceStats.values().stream().mapToLong(
    statsUtilList -> statsUtilList.stream().mapToLong(
        statsUtil -> statsUtil.getResourceInfo().entrySet().stream()
            .filter(e -> !e.getKey().isOnlyForAnalysis())
            .mapToLong(e -> e.getValue().getStatsInfo().get(taskStats).getTotalValue())
            .sum())
        .sum())
    .sum();
```